File: 476. Number Complement.go
package main

func findComplement(num int) int {
	var valid uint32 = 0
	tmp := num
	for (tmp > 0){
		tmp /= 2
		valid ++
	}
	return ^num & ((1<<valid)-1)  // <<操作必须是uint,位取反是^,不是~
}


func findcom(num int) int{
	i := 1
	for (i <= num){
		i = i << 1
	}
	return num ^ (i-1)
}

File: 476. Number Complement.java
public int findComplement(int num) {
    int valid = 0;  // 最高位为1的位数
    int tmp = num;
    while(tmp > 0) {
        tmp /= 2;
        valid++;
    }

    return ~num & ((1 << valid) - 1);
}